pkg/binding/bindingpolicy.go

```diff
@@ -88,6 +88,47 @@ func (c *Controller) syncBindingPolicy(ctx context.Context, bindingPolicyName st
 		logger.V(5).Info("Enqueued Binding for syncing, while handling BindingPolicy", "name", bindingPolicy.Name)
 		c.enqueueBinding(bindingPolicy.GetName())
 
+		statusCollectorName, err := c.identifyMissingStatusCollector(bindingPolicy)
+		if err != nil {
+			return fmt.Errorf("failed to list StatusCollector %s for BindingPolicy %s: %w", statusCollectorName, bindingPolicyName, err)
```

Comment:
I see no list operation here. If you insist on wrapping the error, the wrap should talk about this call site is doing.